File: src/solutions/2310_sum-of-numbers-with-units-digit-k.rs
#[allow(unused)]
use crate::stubs::*;

struct Solution;

// start_submission
impl Solution {
    pub fn minimum_numbers(mut num: i32, k: i32) -> i32 {
        if k == 0 {
            return if num % 10 == 0 {
                if num != 0 { 1 } else { 0 }
            } else {
                -1
            };
        }

        let mut c = 0;

        while num >= 0 {
            if num % 10 == 0 && (num == 0 || c != 0) {
                return c;
            }

            num -= k;
            c += 1;
        }

        -1
    }
}
// end_submission

File: src/cpp_solutions/2310_sum-of-numbers-with-units-digit-k.cpp
#include <bits/stdc++.h>

using namespace std;

// start_submission
class Solution {
public:
    int minimumNumbers(int num, int k) {
        if (num == 0) {
            return 0;
        }

        if (k == 0) {
            return num % 10 == 0 ? 1 : -1;
        }

        int cnt = 0;

        while (num >= 0) {
            if (cnt != 0 && num % 10 == 0) {
                return cnt;
            }

            num -= k;
            cnt++;
        }

        return -1;
    }
};
// end_submission
